libtransact/src/workload/error.rs

```diff
@@ -0,0 +1,55 @@
+use std;
+use std::error::Error as StdError;
+
+use crate::batch::BatchBuildError;
+use crate::transaction::TransactionBuildError;
+
+#[derive(Debug)]
+pub enum WorkloadError {
+    // Returned when an error occurs when using BatchBuilder.
+    BatchBuildError(Box<BatchBuildError>),
+
+    // Returned when an error occurs when using TransactionBuilder.
+    TransactionBuildError(Box<TransactionBuildError>),
+}
+
+impl std::error::Error for WorkloadError {
+    fn description(&self) -> &str {
+        match *self {
+            WorkloadError::BatchBuildError(ref err) => err.description(),
+            WorkloadError::TransactionBuildError(ref err) => err.description(),
+        }
+    }
+
+    fn cause(&self) -> Option<&std::error::Error> {
```

Comment:
Consider implementing this via `source` (see https://doc.rust-lang.org/std/error/trait.Error.html#method.source), and dropping the above description, as it's "Soft-deprecated"

Reply:
I assume you mean dropping cause, not description?

Reply:
Replace cause() with source().